pkg/client/typed_client.go

```diff
@@ -40,9 +40,15 @@ func (c *typedClient) Create(ctx context.Context, obj Object, opts ...CreateOpti
 		return err
 	}
 
+	clusterName := o.GetClusterName()
+	if clusterName == "" {
+		clusterName, _ = ctx.Value("clusterName").(string)
+	}
+
 	createOpts := &CreateOptions{}
 	createOpts.ApplyOptions(opts)
 	return o.Post().
+		Cluster(clusterName).
```

Comment:
Instead of doing all of this to inject the cluster name here, see if instead we can supply a custom `*http.Client` to some Options struct, and in pkg/client/apiutil/apimachinery.go near `RESTClientForGVK`, add a new method that is `RESTClientForGVKAndClient`. That way we can plumb through the kcp custom round tripper.